src/types/NodeType.ts

```diff
@@ -0,0 +1,343 @@
+/* eslint-disable no-underscore-dangle */
```

Comment:
I would say this file took the longest and was pretty robotic about the types. Please make sure that this is all correct. Or commit to this branch if you see anything obvious. I had to add a lot of types here and was confused why they weren't being inferenced

Reply:
well this was the module with all the dumb/fancy logic so it makes sense :p